src/scripts/main.js

```diff
@@ -1,3 +1,316 @@
+};
+
+const notifications = {
+  wrong: {
+    title: 'WRONG!',
+    description: 'All form fields must be completed',
+    type: 'error',
+  },
+  wrongName: {
+    title: 'WRONG NAME!',
+    description: 'The name must contain 4 to 30 letters!',
+    type: 'error',
+  },
+  wrongPosition: {
+    title: 'WRONG POSITION NAME!',
+    description: 'The position name should have more than 2 letters!',
+    type: 'error',
+  },
+  wrongAge: {
+    title: 'WRONG AGE!',
+    description: 'The employee must be over 18 and under 90!',
+    type: 'error',
+  },
+  wrongSalary: {
+    title: 'WRONG SALARY!',
+    description: 'The salary should be a positive number!',
+    type: 'error',
+  },
+  success: {
+    title: 'SUCCESS!',
+    description: 'A new employee was added to the table!',
+    type: 'success',
+  },
+};
+
+const form = document.querySelector('.new-employee-form');
+const formInputs = form.querySelectorAll('input');
+const nameInput = form.querySelector('[data-qa="name"]');
+const position = form.querySelectorAll('[data-qa="position"]');
+const age = form.querySelector('[data-qa="age"]');
+const salary = form.querySelector('[data-qa="salary"]');
+const salaryReg = /^\d{1,}$/;
+const nameReg = /^([a-z]+( [a-z]+)?){4,30}$/i;
+const positionReg = /^([a-z]+( [a-z]+)?){2,50}$/i;
+
+form.addEventListener('submit', function(e) {
+  e.preventDefault();
+
+  const tr = document.createElement('tr');
+  const cols = ['name', 'position', 'office', 'age', 'salary'];
+  const nameVal = nameInput.value;
+  const positionVal = position.value;
+  const ageVal = parseInt(age.value);
+  const salaryVal = salary.value;
+
+  for (let i = 0; i < formInputs.length; i++) {
+    if (!formInputs[i].value) {
+      pushNotification(notifications.wrong);
+
+      return;
+    }
+  }
+
+  if (!nameReg.test(nameVal)) {
+    pushNotification(notifications.wrongName);
+
+    return;
+  }
+
+  if (!positionReg.test(positionVal)) {
+    pushNotification(notifications.wrongPosition);
+  }
+
+  if (ageVal <= 18 || ageVal >= 90) {
+    pushNotification(notifications.wrongAge);
+
+    return;
+  }
+
+  if (!salaryReg.test(salaryVal)) {
+    pushNotification(notifications.wrongSalary);
+
+    return;
+  }
+
+  for (let q = 0; q < cols.length; ++q) {
+    const tdCell = document.createElement('td');
+
+    tdCell.textContent = document.getElementsByName(cols[q])[0].value;
+
+    if (cols[q] === 'salary') {
+      tdCell.textContent = `$`
+        + document.getElementsByName(cols[q])[0].value
+          .toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+    }
+    tr.appendChild(tdCell);
+  }
+
+  document.querySelector('table tbody').appendChild(tr);
+
+  pushNotification(notifications.success);
+
+  document.querySelector('form').reset();
+});
+
+table.addEventListener('dblclick', (e) => {
+  const td = e.target.closest('td');
+
+  if (!td) {
+    return;
+  };
+
+  if (!table.contains(td)) {
+    return;
+  };
+
+  editor(td);
+});
+
+function editor(cell) {
```

Comment:
It's better to name any function with a verb. 
Here it may be `editCell` for example